### ISSUE TITLE
Fails when trying to pass a Request Body with exp_params using the DSL

### ISSUE THREAD

**Issue by JuanFdS** [zhandao/zero-rails_openapi#45]
How to reproduce:
- In a controller, set a request body like this (I left the data and examples empty because they don't matter) :
```
class DogsController < ApplicationController
      api :dogs, 'POST create a doggie' do
         request_body :req, :json, desc: '', data: {}, examples: {}, exp_params: :all
      end
end
```

Error is:
```
rails aborted!
NameError: undefined local variable or method `schema_type' for {}:OpenApi::DSL::MediaTypeObj
Did you mean?  schema
```

I think the problem is here on `media_type_obj.rb`:

```
class MediaTypeObj < Hash
  attr_accessor :media_type, :schema, :examples
      
    def initialize(media_type, hash)
      examples_hash   = hash.delete(:examples)
      exp_params      = schema_type.keys if (exp_params = hash.delete(:exp_params)) == :all
      self.examples   = ExampleObj.new(examples_hash, exp_params, multiple: true) if examples_hash.present?
      self.media_type = media_type_mapping(media_type)
      self.schema     = SchemaObj.new(hash.values_at(:type, :data).compact.first,
      hash.except(:type, :data))
end
```
      

`schema_type` is used to get the `exp_params` but it's not declared anywhere in the class.

Btw, if the bug report is correct, are you open to PRs? because I could create one for this change if you are ok with that.

**Comment by zhandao** [comment]
Thanks, it's a bug, I did something wrong when refactoring:

https://github.com/zhandao/zero-rails_openapi/commit/7ab25a504277d0c782496f5d9e0fa69728fb0bbb#diff-8d475dfd2b68a8aedd99c1048f72459dL15

Welcome to make a PR ~

**Comment by zhandao** [comment]
Hello, it has been fixed by v2.1.1.
